Mathlib/Data/PNat/Basic.lean

```diff
@@ -335,6 +335,23 @@ theorem sub_coe (a b : ℕ+) : ((a - b : ℕ+) : ℕ) = ite (b < a) (a - b : ℕ
     rfl
 #align pnat.sub_coe PNat.sub_coe
 
+theorem sub_le {a b : ℕ+} : a - b ≤ a := by
+  rw [← coe_le_coe, sub_coe]
+  split_ifs with h
+  · exact Nat.sub_le a b
+  · exact a.2
+
+theorem le_sub_one_of_lt {a b : ℕ+} (hab: a < b) : a ≤ b - (1 : ℕ+) := by
+  rw [← coe_le_coe, PNat.sub_coe]
+  split_ifs with h
+  · apply Nat.le_pred_of_lt hab
+  · apply not_lt.mp at h
+    simp_all only [le_one_iff, not_lt_one]
+
+theorem le_of_le_sub_one {a b : ℕ+} (hab: a ≤ b - 1): a ≤ b := by
+  apply LE.le.trans hab (a := a) (c := b) (b := b - 1)
+  exact sub_le
```

Comment:
I'm not sure in which context this would be worth defining: it doesn't seem more than the sum of its parts, and every definition unfortunately carries with its a maintenance cost so we can't add arbitrary corollaries. Could you give an example of where you want to use it?

Reply:
In particular, using previous results we can simplify the proof down to:
```suggestion
theorem le_of_le_sub_one {a b : ℕ+} (hab: a ≤ b - 1): a ≤ b :=
  hab.trans (sub_le _ _)
```
And I'm pretty sure a tactic like `linarith` could solve this immediately (if that is available).

Reply:
> I'm not sure in which context this would be worth defining: it doesn't seem more than the sum of its parts, and every definition unfortunately carries with its a maintenance cost so we can't add arbitrary corollaries. Could you give an example of where you want to use it?

I will remove it. Thanks for the review! These lemmata were the base of a proof, but I'm going a different path now, so you just pick what seems useful.